lib/xterm.sh

```diff
@@ -0,0 +1,32 @@
+# script taken straight from bash-it with almost no adaptation
+# just commented 3 lines and added bash-preexec.sh into .bashrc
+
+#cite about-plugin
+#about-plugin 'automatically set your xterm title with host and location info'
+
+_short-dirname () {
+    local dir_name=$(dirs +0)
+    [ "$SHORT_TERM_LINE" = true ] && [ ${#dir_name} -gt 8 ] && echo "${dir_name##*/}" || echo "$dir_name"
+}
+
+_short-command () {
+    local input_command="$@"
+    [ "$SHORT_TERM_LINE" = true ] && [ ${#input_command} -gt 8 ] && echo "${input_command%% *}" || echo "$input_command"
+}
+
+set_xterm_title () {
+    local title="$1"
+    echo -ne "\033]0;$title\007"
+}
+
+precmd () {
```

Comment:
This overwrites the existing precmd/preexec hooks set by users if any. Instead, the function should be registered to the arrays provided by bash-preexec.

Reply:
fixed